h2o-core/src/main/java/water/rapids/ast/prims/assign/AstRectangleAssign.java

```diff
@@ -66,7 +66,7 @@ public ValFrame apply(Env env, Env.StackHelp stk, AstRoot[] asts) {
       if (rows.isEmpty()) rows = new AstNumList(0, dst.numRows()); // Empty rows is really: all rows
       switch (vsrc.type()) {
         case Val.NUM:
-          assign_frame_scalar(dst, cols, rows, vsrc.getNum(), env._ses);
+          assign_frame_scalar(dst, cols, rows, nanToNull(vsrc.getNum()), env._ses);
```

Comment:
The problem with NA's is that they are always represented as Double.NaN regardless of the type (including UUIDs and Strings).

For the purposes of this change I want the Java type to be compatible with the target H2O type.  Statements like "src instanceof Number" are good example why using NaN is not optimal because it NaN case doesn't really mean src is meant to be a number.

The case you are referring to is not an issue, the cost of this approach is that NA will not pass this test and will be handled by the MR task instead of the shortcut of using constant Vec.